onnx/defs/shape_inference.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "data_type_utils.h"
+
+namespace ONNX_NAMESPACE {
+
+typedef TensorShapeProto_Dimension InferenceDimension;
+
+struct InferenceContext {
```

Comment:
Please kindly keep this interface in the schema.h, so that it's the only thing that other partners (say, me) will need to depend on.